onnxruntime/test/providers/cpu/math/matmul_test.cc

```diff
@@ -352,6 +353,42 @@ TEST(MathOpTest, MatMul_bfloat16) {
 }
 #endif
 
+#if defined(USE_CUDA)
+TEST(MathOpTest, MatMul_float8E4M3FN) {
+  int min_cuda_architecture = 900;
+  if (!HasCudaEnvironment(min_cuda_architecture)) {
+    LOGS_DEFAULT(WARNING) << "Hardware does NOT support Float8E4M3FN";
+    return;
+  }
+  OpTester test("MatMul", 13);
+
+  // TODO add a unit test that has more than 256 elements, so that multiple blocks are used
+  // test.AddInput<MLFloat16>("A", {2, 4}, FloatsToMLFloat16s({1.0f, 2.0f, 3.0f, 4.0f, -1.0f, -2.0f, -3.0f, -4.0f}));
+  // test.AddInput<MLFloat16>("B", {4, 3}, FloatsToMLFloat16s({1.f, 1.f, 1.f, 1.f, 1.f, 1.f, 1.f, 1.f, 1.f, 1.f, 1.f, 1.f}));
+  // test.AddOutput<MLFloat16>("Y", {2, 3}, FloatsToMLFloat16s({10.0f, 10.0f, 10.0f, -10.0f, -10.0f, -10.0f}));
+
+  test.AddInput<MLFloat16>("A", {2, 2}, FloatsToMLFloat16s({1.0f, 1.0f, 1.0f, 1.0f}));
```

Comment:
For FP8 GEMM, pointers and matrix dimension must support 16-byte alignment. 

Could you test input like {2, 16} instead of {2, 2}.

Reply:
Tried that as well, but see similar differences between actual and expected results

```
The difference between f_expected[i] and f_actual[i] is 11.55078125, which exceeds tolerance, where
f_expected[i] evaluates to 16,
f_actual[i] evaluates to 4.44921875, and
tolerance evaluates to 0.018500000238418579.
```

Reply:
I saw that you change A to {2, 16}, but B and output are not 16-byte alignment.

Reply:
I am seeing a similar behavior for M=16, K=32, N=16 as well.

```
The difference between f_expected[i] and f_actual[i] is 7.1015625, which exceeds tolerance, where
f_expected[i] evaluates to 16,
f_actual[i] evaluates to 8.8984375, and
tolerance evaluates to 0.018500000238418579.
```
> We need add some checks before enabling fp8. If requirements are not satisfied, we shall not use fp8.

sure, we can add this